tasks/install_shell.sh

```diff
@@ -350,17 +383,31 @@ do_curl() {
 # do_fetch URL FILENAME
 do_fetch() {
   info "Trying fetch..."
-  fetch -o "$2" "$1" 2>$tmp_stderr
-  # check for bad return status
-  test $? -ne 0 && return 1
+  run_cmd "fetch -o '$2' '$1' 2>$tmp_stderr"
+  rc=$?
+
+  # check for 404
+  grep "404 Not Found" $tmp_stderr 2>&1 >/dev/null
+  if test $? -eq 0; then
+    critical "ERROR 404"
+    unable_to_retrieve_package
+  fi
+
+  # check for bad return status or empty output
+  if test $rc -ne 0 || test ! -s "$2"; then
+    capture_tmp_stderr "fetch"
+    return 1
+  fi
+
   return 0
 }
 
 # do_perl URL FILENAME
 do_perl() {
   info "Trying perl..."
-  perl -e 'use LWP::Simple; getprint($ARGV[0]);' "$1" > "$2" 2>$tmp_stderr
+  run_cmd "perl -e 'use LWP::Simple; getprint($ARGV[0]);' '$1' > '$2' 2>$tmp_stderr"
```

Comment:
$ARGV might get interpolated by bash here since the outer quotes are now double

Reply:
Awesome catch! Tested and updated.

Reply:
You're breathtaking!